### PR TITLE
Add `build-test-publish-wheel` CI file

### PR DESCRIPTION
**Pull request by sarahyurick**
To be merged after https://github.com/NVIDIA/NeMo-Curator/pull/352.

Reference file: https://github.com/NVIDIA/NeMo/blob/main/.github/workflows/build-test-publish-wheel.yml

### CODE REVIEW

File: .github/workflows/build-test-publish-wheel.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build, test, and publish a PyPi wheel (to testpypi)
+
+on:
+  push:
```
Comment (ko3n1g) [inline .github/workflows/build-test-publish-wheel.yml:18]:
```suggestion
  pull_request:
  push:
```

Shall we quickly test this workflow before merging into main? We'd revert this change if everything works

File: .github/workflows/build-test-publish-wheel.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+jobs:
+  build-test-publish-wheel:
+    uses: NVIDIA/NeMo-FW-CI-templates/.github/workflows/_build_test_publish_wheel.yml@v0.7.0
+    with:
+      image-name: nemo_curator_container
+      dockerfile: Dockerfile
+      image-label: nemo-curator
+      build-args: |
+        IMAGE_LABEL=nemo-curator
```
Comment (ko3n1g) [inline .github/workflows/build-test-publish-wheel.yml:36]:
```suggestion
      build-args: |
        IMAGE_LABEL=nemo-curator
        REPO_URL=https://github.com/${{ github.repository }}.git
        CURATOR_COMMIT=${{ github.sha }}
```

**Review by ko3n1g — CHANGES_REQUESTED** [review]
These build args are missing

**Comment by sarahyurick** [comment]
Closing in favor of https://github.com/NVIDIA/NeMo-Curator/pull/356.
